velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -390,6 +394,11 @@ class S3WriteFile::Impl {
 
       part.SetPartNumber(uploadState_.partNumber);
       part.SetETag(result.GetETag());
+      // Don't add the checksum to the part if the checksum is empty.
+      // Some filesystems such as IBM COS require this to be not set.
+      if (!result.GetChecksumCRC32()) {
```

Comment:
Needs `.empty()` based on the build failure and I think we had used this in the internal code.